kolibri/core/content/views.py

```diff
@@ -319,3 +322,30 @@ def get(self, request, filename, new_filename):
         response["Content-Length"] = os.path.getsize(path)
 
         return response
+
+
+class ContentPermalinkRedirect(View):
+
+    def get(self, request, *args, **kwargs):
+
+        # extract the GET parameters
+        channel_id = request.GET.get("channel_id")
+        node_id = request.GET.get("node_id")
+        content_id = request.GET.get("content_id")
+
+        try:  # first, try to get the node by the unique node_id
+            node = ContentNode.objects.get(id=node_id)
```

Comment:
Using `id__startswith` instead of `id` here would allow for partial-hash URLs (e.g. using only first six characters of the node_id as the key).

Same goes for lookups on line `339` and `341`.

Reply:
That could be good! In practice, might be good to always use at least 8, but could make the URLs less massive.

Reply:
In the absence of a node_id param, would need to make sure this is `""` and note `None`, though, so it doesn't blow up

Reply:
Wouldn't starts with "" make it return everything though?

Reply:
Mmm, yeah, so it would blow up anyway... Should probably add an `if` before this section?